server/protocols/jmap/src/main/java/org/apache/james/jmap/utils/SortConverter.java

```diff
@@ -0,0 +1,91 @@
+/****************************************************************
+ O * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.utils;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.james.mailbox.model.SearchQuery;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+public class SortConverter {
+
+    private static final String SEPARATOR = " ";
+    private static final String DESC_ORDERING = "desc";
+    private static final String ASC_ORDERING = "asc";
+
+    @SuppressWarnings("rawtypes")
+    private static final Map<String, SearchQuery.Sort.SortClause> sortClauseMap = ImmutableMap.of(
+        "date", SearchQuery.Sort.SortClause.Arrival,
+        "id", SearchQuery.Sort.SortClause.Uid);
```

Comment:
1/ The team wrote memory sorting like this... I just transfer it
2/ Because I'm nice I have a commit for that later.